streams/src/main/java/org/apache/kafka/streams/processor/StandbyUpdateListener.java

```diff
@@ -0,0 +1,79 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+import org.apache.kafka.common.TopicPartition;
+
+public interface StandbyUpdateListener {
+
+    enum SuspendReason {
+        MIGRATED,
+        PROMOTED
+    }
+
+    /**
+     * Method called upon the creation of the Standby Task.
+     *
+     * @param topicPartition   the TopicPartition of the Standby Task.
+     * @param storeName        the name of the store being watched by this Standby Task.
+     * @param startingOffset   the offset from which the Standby Task starts watching.
+     */
+    void onUpdateStart(final TopicPartition topicPartition,
+                       final String storeName,
+                       final long startingOffset);
+
+    /**
+     * Method called after restoring a batch of records. In this case the maximum size of the batch is whatever
+     * the value of the MAX_POLL_RECORDS is set to.
+     * <n>
+     * This method is called after restoring each batch and it is advised to keep processing to a minimum.
+     * Any heavy processing will hold up recovering the next batch, hence slowing down the restore process as a
+     * whole.
+     *
+     * If you need to do any extended processing or connecting to an external service consider doing so asynchronously.
+     *
+     * @param topicPartition the TopicPartition containing the values to restore
+     * @param storeName the name of the store undergoing restoration
+     * @param batchEndOffset the inclusive ending offset for the current restored batch for this TopicPartition
+     * @param numRestored the total number of records restored in this batch for this TopicPartition
+     * @param currentEndOffset the current end offset of the changelog topic partition.
+     */
+    void onBatchLoaded(final TopicPartition topicPartition,
+                       final String storeName,
+                       final TaskId taskId,
+                       final long batchEndOffset,
+                       final long numRestored,
+                       final long currentEndOffset);
+
+    /**
+     * Method called after a Standby Task is closed, either because the Standby Task was promoted to an Active Task
+     * or because the Standby Task was migrated to another instance (in which case the data will be cleaned up
+     * after state.cleanup.delay.ms).
```

Comment:
```suggestion
     * This method is called when the corresponding standby task stops updating, for the provided reason.
     * <p>
     * If the task was {@code MIGRATED} to another instance, this callback will be invoked after this
     * state store (and the task itself) are closed (in which case the data will be cleaned up after 
     * state.cleanup.delay.ms).
     * If the task was {@code PROMOTED} to an active task, the state store will not be closed, and the 
     * callback will be invoked after unregistering it as a standby task but before re-registering it as an active task 
     * and beginning restoration. In other words, this will always called before the corresponding 
     * {@link StateRestoreListener#onRestoreStart} call is made.
```